src/core/AutoTypeAssociations.cpp

```diff
@@ -115,3 +115,16 @@ void AutoTypeAssociations::clear()
 {
     m_associations.clear();
 }
+
+bool AutoTypeAssociations::operator==(const AutoTypeAssociations& other) const
+{
+    if (m_associations.count() != other.m_associations.count()) {
+        return false;
+    }
+    for (int i = 0; i < m_associations.count(); ++i) {
+        if (m_associations[i] != other.m_associations[i]) {
```

Comment:
This checks that two lists share the "same" elements in an order-preserving comparison, but I don't think that this is what we want here.

An AutotypeAssociations is a list of Association like `[ Association1, Association2, Association3 ]`.
I think the comparison should be order-less.

Should the above example be considered **==** to `[ Association2, Association3, Association1 ]` ? What do you guys think?

Reply:
In my opinion the above example should only be considered equal when there is no code which depends on the order of the associations. Since it is about auto typing, I assume that is or there may be some feature which does execute the first association on after pressing shortcut - in this case we have to compare order preserving.